### PR TITLE
Update accAPI.cpp to allow multiple uda servers connection

### PR DESCRIPTION
**Pull request by DorisBDR**
retrieve the server host and port from environment to have a proper comaprison

### CODE REVIEW

File: source/client/accAPI.cpp [hunk 1]
```diff
@@ -702,7 +702,7 @@ void putIdamServerPort(int port)
 {
     ENVIRONMENT* environment = getIdamClientEnvironment();
     int old_port = port;
-    environment->server_port = port;                             // UDA server service port number
+    environment->server_port = environment->server_port;                             // UDA server service port number
```
[the remark this answers is not in the document]
Comment (DorisBDR) [reply source/client/accAPI.cpp:705]: sure i was not fully wake up this mornign. thanks

**Comment by jholloc** [comment]
I've incorporated these changes in the the release/2.8.0 branch. Please use this branch for any future testing.

**Comment by DorisBDR** [comment]
i found another issue...this one is a little bit tricky to solve but very annoying for us. if we have a new client (e.g. MINT) and two uda servers one new and one old then we can get data because there is an issue with the protocol_version...which is not correct (it thinks that the uda server is new whereas it is an old one).

**Comment by DorisBDR** [comment]
commits done on our side to support multiple servers (including different versions), waiting for the build to check that things are ok

**Comment by olivhoenen** [comment]
@jholloc can you please look at bringing in https://github.com/ukaea/UDA/pull/19/commits/7e6d5c065ac3f877d018cbe79dba62e7dbb9995e and https://github.com/ukaea/UDA/pull/19/commits/5255a1bf8ed49963ce8b77f4edc85574d5337a00 so we can get closer to a 2.8.0 release? 

I get that this PR may no target the correct branch (`main`, shall probably be `develop` or `release/2.8.0` instead, --> will need to be clarified in guidelines for developers in the future).

**Comment by jholloc** [comment]
This has been merged into release/2.8.0

**Comment by jholloc** [comment]
This has been merged into release/2.8.0

**Pull request closed without merging**